index.js

```diff
@@ -25,9 +25,17 @@ Document.prototype.setInputString = function (string, bufferSize) {
 }
 
 ASTNodeArray.prototype[Symbol.iterator] = function* () {
-  let i = 0;
-  while(i < this.length) {
-    yield this[i++];
+  let node = this[0];
+
+  const getNext = this.isNamed ?
+    (node) => node.nextNamedSibling :
+    (node) => node.nextSibling;
+
+  if (node) {
+    yield node;
+    while ((node = getNext(node))) {
+      yield node;
+    }
```

Comment:
Even though it will mean more duplication, I'd prefer to have one conditional check at the beginning of the loop, rather than a closure with a conditional on each iteration. Something like:

```js
if (this.isNamed) {
  while ((node = node.nextNamedSibling)) {
    yield node
  }
} else {
  while ((node = node.nextSibling)) {
    yield node
  }
}
```

Reply:
I can do this, but I wanted to mention that the way it's currently implemented doesn't use a closure and the `this.isNamed` is only evaluated once. The only difference this change would make would be removing the stack frame of the lambda on lines 31/32.

Reply:
Ah, thanks for pointing that out; I misread your code.